src/Bladeburner/Actions/Action.ts

```diff
@@ -146,9 +146,19 @@ export abstract class ActionClass {
     let high = real + diff;
     const city = bladeburner.getCurrentCity();
     let r = city.pop / city.popEst;
-    if (Number.isNaN(r)) r = 0;
-    if (r < 1) low *= r;
-    else high *= r;
+    if (Number.isNaN(r)) {
+      r = 0;
+    }
+    if (r < 1) {
+      low *= r;
+    } else {
+      // This happens when the action is Raid, popEst=0, and comms=0.
```

Comment:
I think you can simplify this to just `high *= clamp(r)`, which is easier to reason about.

Reply:
Do you mean to simplify the entire `if (high === 0 && r === Infinity) { high = 0; } else { high *= r; }` to `high *= clamp(r)`?
I don't think we can do that. `r` can be greater than 1. For example, if `city.pop` is 2x`city.popEst`, `r` is 2.
We may use the `clampNumber` function: `high *= clampNumber(r)`.

Reply:
Ah yes, that is what I meant. The new function Caldwell added for dealing with this type of issue.